From 2c02d1cee90bbf1be9b9b2af35d996307476b336 Mon Sep 17 00:00:00 2001
From: zhang defang <defang.zhang@spreadtrum.com>
Date: Fri, 12 Oct 2018 10:34:01 +0800
Subject: [PATCH 2/2] fGpu is null when GrGpuResource::release

[root cause]
fGpu is null when GrGpuResource::release,call get_resource_cache(fGpu)->resourceAccess().removeResource(this);
happened native crash,like GrGpuResource::abandon(),add condition wasDestroyed protected

[changes ]
add condition wasDestroyed protected

Change-Id: I9d97e08aae3656e473dd4ce1fe40722f45948fa8
Signed-off-by: Pranav Vashi <neobuddy89@gmail.com>
Signed-off-by: DennySPB <dennyspb@gmail.com>
---
 src/gpu/GrGpuResource.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/gpu/GrGpuResource.cpp b/src/gpu/GrGpuResource.cpp
index ec7113562b..a10cf06ada 100644
--- a/src/gpu/GrGpuResource.cpp
+++ b/src/gpu/GrGpuResource.cpp
@@ -50,6 +50,9 @@ GrGpuResource::~GrGpuResource() {
 }
 
 void GrGpuResource::release() {
+    if (this->wasDestroyed()) {
+        return;
+    }
     SkASSERT(fGpu);
     this->onRelease();
     get_resource_cache(fGpu)->resourceAccess().removeResource(this);
-- 
2.21.0

